### PR TITLE
Added static Font var:$font_space and static method setFontSpaceLimit to have an opportunity to set custom value.

### LINKED ISSUE #259: getFontSpaceLimit - class variable?
Is there any chance you could make this a configurable class variable?  Dirtyish quick way would be to set it statically on the Font object before you create a Parser instance and away we go?

At the moment we have to manually change the file every time we update the package in composer and we can't really extend the classes efficiently to fix it that way.

### PR DESCRIPTION
**Pull request by butschster**
…mit` to have an opportunity to set custom value.

**Example**
```php
\Smalot\PdfParser\Font::setFontSpaceLimit(-100);

$oParser = new \Smalot\PdfParser\Parser();
$oPDF = $oParser->parseFile($sFile2Convert);
$sRawFile = $oPDF->getText();
```

Closes #259

### CODE REVIEW

**Review by k00ni — COMMENTED** [review]
Thank you for your contribution! At first I found it very handy to set font space limit using a "central" point. 

But on the contrary I see the following downside. In my opinion (and based on some blog posts) using singletons (and static functions) is problematic, because you define them globally and force depending code to use changed values. In case you forget it or if someone else changes it during a process, you may end up with race conditions and inconsistency. Changing and forgetting about it is a risk and may affect parsing processes, especially if there are further values which may vary between different PDFs.

Did you read my issue #305? What do you think about the idea to move such "config" to a separate object which is provided at the beginning of the parsing process?

File: tests/Integration/FontTest.php [hunk 1]
```diff
@@ -40,6 +40,15 @@
 
 class FontTest extends TestCase
 {
+    public function testSetFontSpaceLimit()
+    {
+        $this->assertEquals(-50, Font::getFontSpaceLimit());
+        Font::setFontSpaceLimit(-100);
+        $this->assertEquals(-100, Font::getFontSpaceLimit());
+
+        Font::setFontSpaceLimit(-50);
```
[the remark this answers is not in the document]
Comment (butschster) [reply tests/Integration/FontTest.php:49]: I couldn't run tests on a local machine, I don't have php5, but there was maybe problem with this line for next tests and I decided to add it. I can remove it and we will check tests.

[the remark this answers is not in the document]
Comment (k00ni) [reply tests/Integration/FontTest.php:49]: > but there was maybe problem with this line for next tests and I decided to add it.

That's a bad sign, because tests have to be independent and isolated. Either try to check `setUp` and/or `tearDown` functions or the implementation itself.

Regarding PHP5: You can change your code and commit as often as you want. Afterwards our test environment gets triggered and you will see the results below. Don't be afraid to check for edge cases etc.

[the remark this answers is not in the document]
Comment (butschster) [reply tests/Integration/FontTest.php:49]: I'm trying today to get rid of this line and test on a local machine, it's not a problem! I think according to current release this fix is the best way not to break projects using this package. But I like the idea to use config. I also don't like singletons and want to have clear code, that will have predictable behavior, but in this case, maybe, this is the only way without breaking everything. Maybe configuration from config is a good idea for v.1.0

**Comment by k00ni** [comment]
@j0k3r and @Connum any feedback on this PR? Are you +1 for the singleton solution or more for the Config approach?

**Comment by Connum** [comment]
+1 for config, seems more flexible for the future and allows for multiple instances with different values.

**Comment by j0k3r** [comment]
I agree about putting it in config

**Comment by k00ni** [comment]
@butschster I am sorry for the late response. 

I created #375 which contains a non-singleton approach to set font space limit before parsing a PDF. Would love to hear your opinion on this. I prefer the non-singleton approach for the reasons I described earlier. Hope we can make #375 work.

**Comment by butschster** [comment]
> @butschster I am sorry for the late response.
> 
> I created #375 which contains a non-singleton approach to set font space limit before parsing a PDF. Would love to hear your opinion on this. I prefer the non-singleton approach for the reasons I described earlier. Hope we can make #375 work.
> 
> Thank you for your effort though!

Hi, @k00ni! 
I've seen your PR and this is very good solution. I'm glad, that my PR helped to start working on better solution than my fix :)

Thank you!
